cloud/pkg/synccontroller/objectsync.go

```diff
@@ -34,38 +32,26 @@ func (sctl *SyncController) manageObject(sync *v1alpha1.ObjectSync) {
 
 	//ret, err := informers.GetInformersManager().GetDynamicSharedInformerFactory().ForResource(gvr).Lister().ByNamespace(sync.Namespace).Get(sync.Spec.ObjectName)
 	ret, err := sctl.kubeclient.Resource(gvr).Namespace(sync.Namespace).Get(context.TODO(), sync.Spec.ObjectName, metav1.GetOptions{})
-	if err != nil {
+	if err != nil || ret == nil {
 		klog.Errorf("failed to get obj(gvr:%v,namespace:%v,name:%v), %v", gvr, sync.Namespace, sync.Spec.ObjectName, err)
 		return
 	}
 
 	nodeName := getNodeName(sync.Name)
-	if ret != nil {
-		object, err = meta.Accessor(ret)
-		if err != nil {
-			return
-		}
-
-		syncObjUID := getObjectUID(sync.Name)
-		if syncObjUID != string(object.GetUID()) {
-			err = apierrors.NewNotFound(schema.GroupResource{
-				Group:    "",
-				Resource: sync.Spec.ObjectKind,
-			}, sync.Spec.ObjectName)
-		}
-	}
 
+	object, err = meta.Accessor(ret)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			obj := &unstructured.Unstructured{}
-			obj.SetName(sync.Spec.ObjectName)
-			obj.SetUID(types.UID(getObjectUID(sync.Name)))
-			obj.SetNamespace(sync.Namespace)
-		} else {
-			klog.Errorf("Failed to manage pod sync of %s in namespace %s: %v", sync.Name, sync.Namespace, err)
-			return
-		}
```

Comment:
this `if` branch is unnecessary, `err != nil` happenes only when the `syncObjUID != string(object.GetUID())` and the err is always `notfound`, and here define a not-used `Unstructured` object, so we can delete these codes.